### DISCUSSION TITLE
decouple postgres and redis ?

### DISCUSSION THREAD

**Discussion by meokey** [outline/outline#7940] [Self Hosting]
I'm trying to self-host Outline but notice that it have postgres and redis built-in the docker-compose.yml file. I have already the 2 services running so I would like to reuse them and do not want to run the 2 databases again with Outline.
Is it possible to decouple the dependences and use environment variables for the connection strings and other parameters? Maybe also provide an executable script/SQL to initialize the database? Thanks.

Thank you.

**Comment by tommoor** [comment]
Sure, you can set it up however you like. The compose file in the docs is just an example implementation!

**Reply by meokey** [reply]
Thanks for your reply - just to confirm: with my own postgres and redis, I need to 
- use parameter -net=localhost (what if the database is on other server?)
- create an .env file from .env.sample and modify the parameters

and, how can I initialize the database? 
Thanks.
